### PR TITLE
Make region file names snake_case (part 7)

### LINKED ISSUE #3641: Make region file names snake_case
as a part of my snake_case series of issues

### PR DESCRIPTION
**Pull request by rhyolight**
Fixes #3641

Do not merge. Tests are failing. Posting here to show the example.

### CODE REVIEW

File: src/nupic/engine/__init__.py [hunk 1]
```diff
@@ -49,7 +49,7 @@
     ("nupic.regions.KNNClassifierRegion", "KNNClassifierRegion"),
     ("nupic.regions.PluggableEncoderSensor", "PluggableEncoderSensor"),
     ("nupic.regions.PyRegion", "PyRegion"),
-    ("nupic.regions.RecordSensor", "RecordSensor"),
```
Comment (rhyolight) [inline src/nupic/engine/__init__.py]: @vitaly-krugl @scottpurdy Here I have updated the region registration, but still getting a test error:

```
> python tests/unit/nupic/regions/record_sensor_region_test.py
```

```
ERR:  CHECK FAILED: "pClass && PyType_Check(pClass)"  [/Users/mtaylor/nta/nupic.core/src/nupic/py_support/PyHelpers.cpp line 656]
WARN:   PyRegion::createSpec failed: 0x7fff521cddd8
EERR:  CHECK FAILED: "pClass && PyType_Check(pClass)"  [/Users/mtaylor/nta/nupic.core/src/nupic/py_support/PyHelpers.cpp line 656]
WARN:   PyRegion::createSpec failed: 0x7fff521cddd8
EERR:  CHECK FAILED: "pClass && PyType_Check(pClass)"  [/Users/mtaylor/nta/nupic.core/src/nupic/py_support/PyHelpers.cpp line 656]
WARN:   PyRegion::createSpec failed: 0x7fff521cdf48
E
======================================================================
ERROR: testActValueOut (__main__.RecordSensorRegionTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/unit/nupic/regions/record_sensor_region_test.py", line 118, in testActValueOut
    network = _createNetwork()
  File "tests/unit/nupic/regions/record_sensor_region_test.py", line 36, in _createNetwork
    network.addRegion('sensor', 'py.record_sensor', '{}')
  File "/Users/mtaylor/nta/nupic/src/nupic/engine/__init__.py", line 641, in addRegion
    engine_internal.Network.addRegion(self, name, nodeType, nodeParams)
  File "/Users/mtaylor/nta/nupic.core/bindings/py/src/nupic/bindings/engine_internal.py", line 1167, in addRegion
    return _engine_internal.Network_addRegion(self, *args, **kwargs)
AttributeError: 'module' object has no attribute 'record_sensor'

======================================================================
ERROR: testBucketIdxOut (__main__.RecordSensorRegionTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/unit/nupic/regions/record_sensor_region_test.py", line 111, in testBucketIdxOut
    network = _createNetwork()
  File "tests/unit/nupic/regions/record_sensor_region_test.py", line 36, in _createNetwork
    network.addRegion('sensor', 'py.record_sensor', '{}')
  File "/Users/mtaylor/nta/nupic/src/nupic/engine/__init__.py", line 641, in addRegion
    engine_internal.Network.addRegion(self, name, nodeType, nodeParams)
  File "/Users/mtaylor/nta/nupic.core/bindings/py/src/nupic/bindings/engine_internal.py", line 1167, in addRegion
    return _engine_internal.Network_addRegion(self, *args, **kwargs)
AttributeError: 'module' object has no attribute 'record_sensor'

======================================================================
ERROR: testVaryingNumberOfCategories (__main__.RecordSensorRegionTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/unit/nupic/regions/record_sensor_region_test.py", line 72, in testVaryingNumberOfCategories
    "sensor", "py.record_sensor", "{'numCategories': 2}")
  File "/Users/mtaylor/nta/nupic/src/nupic/engine/__init__.py", line 641, in addRegion
    engine_internal.Network.addRegion(self, name, nodeType, nodeParams)
  File "/Users/mtaylor/nta/nupic.core/bindings/py/src/nupic/bindings/engine_internal.py", line 1167, in addRegion
    return _engine_internal.Network_addRegion(self, *args, **kwargs)
AttributeError: 'module' object has no attribute 'record_sensor'

----------------------------------------------------------------------
Ran 3 tests in 0.005s

FAILED (errors=3)
```

Comment (scottpurdy) [inline src/nupic/engine/__init__.py]: I think it should be:

```
("nupic.regions.record_sensor", "RecordSensor"),
```

Comment (rhyolight) [inline src/nupic/engine/__init__.py]: Tried that first, same problem.

Comment (vitaly-krugl) [inline src/nupic/engine/__init__.py]: @rhyolight: @scottpurdy is right. See my PR that demonstrated the same change that you're attempting to make and built successfully in all environments: https://github.com/numenta/nupic/pull/3632/files.

Please compare your changes against my PR or reopen my PR and use it, instead.

File: docs/examples/network/complete-network-example.py [hunk 2]
```diff
@@ -58,7 +58,7 @@ def createNetwork(dataSource):
   network = Network()
 
   # Add a sensor region.
-  network.addRegion("sensor", "py.RecordSensor", '{}')
+  network.addRegion("sensor", "py.record_sensor", '{}')
```
Comment (vitaly-krugl) [inline docs/examples/network/complete-network-example.py]: @rhyolight, this change and othes like it are the culprit. You should leave them alone. "py.RecordSensor" refers to the class name, not the module name. The registration logic then locates the module name that matches the class name.

File: docs/examples/network/example-create-regions.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import json
 
 # Add a sensor region, set its encoder and data source.
-network.addRegion("sensor", "py.RecordSensor", json.dumps({"verbosity": 0}))
+network.addRegion("sensor", "py.record_sensor", json.dumps({"verbosity": 0}))
```
Comment (scottpurdy) [inline docs/examples/network/example-create-regions.py]: This should be `"py.RecordSensor"`

File: examples/network/custom_region_demo.py [hunk 4]
```diff
@@ -68,7 +68,7 @@ def createNetwork(dataSource):
   # Our input is sensor data from the gym file. The RecordSensor region
   # allows us to specify a file record stream as the input source via the
   # dataSource attribute.
-  network.addRegion("sensor", "py.RecordSensor",
+  network.addRegion("sensor", "py.record_sensor",
```
Comment (scottpurdy) [inline examples/network/custom_region_demo.py]: `"py.RecordSensor"`

File: examples/network/hierarchy_network_demo.py [hunk 5]
```diff
@@ -135,7 +135,7 @@ def createRecordSensor(network, name, dataSource):
   """
 
   # Specific type of region. Possible options can be found in /nupic/regions/
-  regionType = "py.RecordSensor"
+  regionType = "py.record_sensor"
```
Comment (scottpurdy) [inline examples/network/hierarchy_network_demo.py]: `"py.RecordSensor"`

File: src/nupic/engine/__init__.py (hunk 1 above)
Comment (scottpurdy) [inline src/nupic/engine/__init__.py]: See other comments on a few of the places that need to be updated to `"py.RecordSensor"`. Can you change this back to my suggestion so I can see everything in context. I'm not sure which permutations you tried.

**Comment by rhyolight** [comment]
@scottpurdy Figured it out, thanks. I changed the wrong strings.

**Comment by rhyolight** [comment]
@scottpurdy @vitaly-krugl Ok guys. I'm going to add the rest of the region name changes to this PR. More commits coming.

**Comment by rhyolight** [comment]
@scottpurdy or @vitaly-krugl This is ready for final review.

**Pull request merged**